ts/Stock/Indicators/IKH/IKHIndicator.ts

```diff
@@ -0,0 +1,960 @@
+            xVal.length <= period ||
+            !isArray(yVal[0]) ||
+            yVal[0].length !== 4
+        ) {
+            return;
+        }
+
+        // Add timestamps at the beginning
+        dateStart = xVal[0] - period * closestPointRange;
+
+        for (i = 0; i < period; i++) {
+            xData.push(dateStart + i * closestPointRange);
+        }
+
+        for (i = 0; i < yValLen; i++) {
+            // Tenkan Sen
+            if (i >= periodTenkan) {
+                slicedTSY = yVal.slice(i - periodTenkan, i);
+
+                pointTS = highlowLevel(slicedTSY);
+
+                TS = (pointTS.high + pointTS.low) / 2;
+            }
+
+            if (i >= period) {
+                slicedKSY = yVal.slice(i - period, i);
+
+                pointKS = highlowLevel(slicedKSY);
+
+                KS = (pointKS.high + pointKS.low) / 2;
+
+                SSA = ((TS as any) + KS) / 2;
+            }
+
+            if (i >= periodSenkouSpanB) {
+                slicedSSBY = yVal.slice(i - periodSenkouSpanB, i);
+
+                pointSSB = highlowLevel(slicedSSBY);
+
+                SSB = (pointSSB.high + pointSSB.low) / 2;
+            }
+
+            CS = yVal[i][3];
+
+            date = xVal[i];
+
+            if (typeof IKH[i] === 'undefined') {
+                IKH[i] = [];
+            }
+
+            if (typeof IKH[i + period] === 'undefined') {
+                IKH[i + period] = [];
+            }
+
+            IKH[i + period][0] = TS;
+            IKH[i + period][1] = KS;
+            IKH[i + period][2] = void 0;
+
+            IKH[i][2] = CS;
+
+            if (i <= period) {
+                IKH[i + period][3] = void 0;
+                IKH[i + period][4] = void 0;
+            }
+
+            if (typeof IKH[i + 2 * period] === 'undefined') {
+                IKH[i + 2 * period] = [];
+            }
+
+            IKH[i + 2 * period][3] = SSA;
+            IKH[i + 2 * period][4] = SSB;
+
+            xData.push(date);
+        }
+
+        // Add timestamps for further points
+        for (i = 1; i <= period; i++) {
+            xData.push((date as any) + i * closestPointRange);
+        }
+
+        return {
+            values: IKH,
+            xData: xData,
+            yData: IKH
+        } as IndicatorValuesObject<TLinkedSeries>;
+    }
+}
+
+/* eslint-disable @typescript-eslint/interface-name-prefix */
+
+/* *
+*
+* Prototype Properties
+*
+* */
+interface IKHIndicator {
+    data: Array<IKHPoint>;
+    getValues<TLinkedSeries extends LineSeries>(
+        series: TLinkedSeries,
+        params: IKHParamsOptions
+    ): IndicatorValuesObject<TLinkedSeries> | undefined;
+    graphCollection: Array<string>;
+    graphsenkouSpan: SVGElement | undefined;
+    ikhMap: Highcharts.Dictionary<Array<IKHPoint>>;
+    nextPoints?: Array<IKHPoint>;
+    options: IKHOptions;
+    pointValKey: string;
+    pointClass: typeof IKHPoint;
+    points: Array<IKHPoint>;
+    pointArrayMap: Array<keyof IKHPoint>;
```

Comment:
Every property here, that will be not part of the extend call (has a default value) needs to be moved to class pattern and in some cases initialized with `void 0 as any`. Exception to this rule is the pointClass property, which stays in the interface.

Reply:
Moved given properties to class pattern